src/installer/managed/Microsoft.NET.HostModel/Bundle/TargetInfo.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.NET.HostModel.AppHost;
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.HostModel.Bundle
+{
+    /// <summary>
+    /// TargetInfo: Information about the target for which the single-file bundle is built.
+    /// 
+    /// Currently the TargetInfo only tracks:
+    ///   - the target operating system
+    ///   - The target framework
+    /// If necessary, the target architecture may be tracked in future.
+    /// </summary>
+
+    public class TargetInfo
+    {
+        public readonly OSPlatform OS;
+        public readonly string Framework;
+        public readonly uint BundleVersion;
+        public readonly BundleOptions DefaultOptions;
+
+        public TargetInfo(OSPlatform? os, string framework)
+        {
+            OS = os ?? HostOS;
+            Framework = framework != null ? framework.ToLower() : "net5";
+
+            Debug.Assert(IsLinux || IsOSX || IsWindows);
+            switch (Framework)
+            {
+                case "netcoreapp3.0":
+                case "netcoreapp3.1":
+                    BundleVersion = 1u;
+                    DefaultOptions = BundleOptions.BundleAllContent;
+                    break;
+
+                case "net5":
```

Comment:
Not sure how the discussion around .NET 5 TFMs ended up (if it already happened), but I vaguely remember that there were mote than one .NET 5 TFM. Will the SDK make sure to always pass "net5" for all such TFMs, or should we support all the other .NET 5 TFMS here?

Reply:
Spec and discussion here: https://github.com/dotnet/designs/pull/92

Yes, there is multiple TFMs, it is better to use the pre-parsed identifiers and versions.

Reply:
Spec and discussion here: https://github.com/dotnet/designs/pull/92

Yes, there is multiple TFMs, it is better to use the pre-parsed identifiers and versions.

Reply:
I wrote down the framework strings to make it evident that the bundle versions are tied to TFMs.  I'll change it to accept the version number only (3, 3.1, 5). This way, the SDK can efficiently pass the `TargetFrameworkNumberWithoutV` as input.

Reply:
Nit: ">= 5" for the last case. :-) I'm sure there is going to be some next version at some point and we want it to default to the new code path.